zubhub_backend/zubhub/activities/tests/test_utils.py

```diff
@@ -0,0 +1,223 @@
+    image_count_after = Image.objects.all().count()
+    # assert that images and activity images were created
+    self.assertEqual(activity_image_count_after - activity_image_count_before, len(images))
+    self.assertEqual(image_count_after - image_count_before, len(images))
+    
+  def test_create_making_steps(self):
+    making_steps = [
+      {
+        'title': 'Step 1',
+        'description': 'Description for step 1',
+        'step_order': 1,
+        'image': [{'file_url': 'https://picsum.photos/710','public_id': '1',}]
+      },
+      {
+        'title': 'Step 2',
+        'description': 'Description for step 2',
+        'step_order': 2,
+        'image': [{'file_url': 'https://picsum.photos/720','public_id': '2',}]
+      },
+      {
+        'title': 'Step 3',
+        'description': 'Description for step 3',
+        'step_order': 3,
+        'image': [{'file_url': 'https://picsum.photos/730','public_id': '3',}]
+      },
+    ]
+    
+    # get counts before calling function
+    activity_steps_count_before = ActivityMakingStep.objects.all().count()
+    image_count_before = Image.objects.all().count()
+    # call function to be tested
+    create_making_steps(self.test_activity, making_steps)
+    # get counts after function call
+    activity_steps_count_after = ActivityMakingStep.objects.all().count()
+    image_count_after = Image.objects.all().count()
+    self.assertEquals(activity_steps_count_after - activity_steps_count_before, len(making_steps))
+    # check that there are now more images than there were before
+    self.assertGreater(image_count_after, image_count_before)
+    
+  def test_create_inspiring_example(self):
+    inspiring_examples_data = [
+      {
+        'description': 'Inspiring example 1',
+        'credit': 'credit',
+        'image': {
+          'file_url': 'https://picsum.photos/300',
+          'public_id': 'A sample image from the loren picsum website',
+        }
+      },
+      {
+        'description': 'Inspiring example 2',
+        'credit': 'credit',
+        # no image in this inspiring example
+      },
+    ]
+    
+    # get number of examples and images before function call
+    examples_count_before = InspiringExample.objects.filter(activity=self.test_activity).count()
+    images_count_before = Image.objects.all().count()
+    # function call
+    create_inspiring_examples(self.test_activity, inspiring_examples=inspiring_examples_data)
+    # get number of examples and images after function call
+    examples_count_after = InspiringExample.objects.filter(activity=self.test_activity).count()
+    images_count_after = Image.objects.all().count()
+    # Assert new InspiringExample instances were created
+    self.assertGreater(examples_count_after, examples_count_before)
+    # Assert only one image was created
+    self.assertEqual(images_count_after - images_count_before, 1)
+
+  def test_update_activity_image(self):
```

Comment:
here you need to create some initial `ActivityImages` since we are testing an update operation.
We also need to mock out `create_activity_images` (this is to make sure that `update_activity_image` test can't be affected when change is made to `create_activity_images`. You can read more about mocking online).

Now after running `update_activity_image`, we check that the old `ActivityImages` that we manually created earlier no longer exists,
and check that the contents of the `images` array were passed to our mock.